.github/copilot-instructions.md

```diff
@@ -0,0 +1,186 @@
+# Advent of Code Solutions Repository
+
+**ALWAYS follow these instructions first and only fallback to additional search and context gathering if the information here is incomplete or found to be in error.**
+
+This repository contains Rust solutions for [Advent of Code](https://adventofcode.com/) problems from 2015-2024. It uses a cargo workspace structure with each year as a separate crate and each day as a separate binary.
+
+## Working Effectively
+
+### Prerequisites
+- Rust and Cargo must be installed
+- Valid Advent of Code session token required for fetching puzzle input
+
+### Essential Commands with Timing
+Execute these commands from the repository root:
+
+**Build the entire workspace:**
+```bash
+cargo build
+```
+- Takes ~47 seconds to complete. **NEVER CANCEL** - set timeout to 90+ minutes.
+- Produces warnings about lifetime elision (this is expected).
```

Comment:
This have been solved by the latest commit, should not happen after solving the new clippy lints